lib/rubocop/cop/performance/count.rb

```diff
@@ -27,56 +30,67 @@ class Count < Cop
         COUNTERS = [:count, :length, :size]
 
         def on_send(node)
-          expression, first_method, second_method, third_method = parse(node)
+          expression, selector, params, counter = parse(node)
+          return unless COUNTERS.include?(counter)
 
-          return unless COUNTERS.include?(third_method)
-
-          begin_pos = if SELECTORS.include?(first_method)
-                        return if second_method.is_a?(Symbol)
-                        expression.loc.selector.begin_pos
+          begin_pos = if selector.is_a?(Symbol)
+                        expression.parent.loc.selector.begin_pos if expression
                       else
-                        return unless SELECTORS.include?(second_method)
-                        expression.parent.loc.selector.begin_pos
+                        _enumerable, selector, params = *expression
+
+                        if contains_selector?(expression)
+                          expression.loc.selector.begin_pos
+                        end
                       end
 
+          return unless SELECTORS.include?(selector)
+          return if params && !params.block_pass_type?
           return if node.parent && node.parent.block_type?
 
+          begin_pos ||= node.loc.expression.begin_pos
+
           range = Parser::Source::Range.new(node.loc.expression.source_buffer,
```

Comment:
Good idea. I have caused the wrong, or poor highlighting in the past.

In general, I think that it would be a good idea for all cops to start have tests for `highlights`.